kubelet/datadog_checks/kubelet/summary.py

```diff
@@ -0,0 +1,149 @@
+
+class SummaryScraperMixin(object):
+    """
+    This class scrapes metrics from Kubelet "/stats/summary" endpoint
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(SummaryScraperMixin, self).__init__(*args, **kwargs)
+
+    def process_stats_summary(self, pod_list_utils, stats, instance_tags, main_stats_source):
+        # Reports system container metrics (node-wide)
+        self._report_system_container_metrics(stats, instance_tags)
+        # Reports POD & Container metrics. If `main_stats_source` is set, retrieve everything it can
+        # Otherwise retrieves only what we cannot get elsewhere
+        self._report_metrics(pod_list_utils, stats, instance_tags, main_stats_source)
+
+    def _report_metrics(self, pod_list_utils, stats, instance_tags, main_stats_source):
+        for pod in stats.get('pods', []):
+            pod_namespace = pod.get('podRef', {}).get('namespace')
+            pod_name = pod.get('podRef', {}).get('name')
+            pod_uid = pod.get('podRef', {}).get('uid')
+
+            if pod_namespace is None or pod_name is None or pod_uid is None:
+                self.log.warning("Kubelet returned incomplete results from '/stats/summary'")
+                continue
+
+            self._report_pod_stats(pod_namespace, pod_name, pod_uid, pod, instance_tags, main_stats_source)
+            self._report_container_stats(pod_namespace, pod_name, pod.get('containers', []), pod_list_utils, instance_tags, main_stats_source)
+
+    def _report_pod_stats(self, pod_namespace, pod_name, pod_uid, pod, instance_tags, main_stats_source):
+        pod_tags = tags_for_pod(pod_uid, tagger.ORCHESTRATOR)
+        if not pod_tags:
+            self.log.debug("Tags not found for pod: %s/%s - no metrics will be sent", pod_namespace, pod_name)
+            return
+        pod_tags += instance_tags
+
+        pod_ephemeral_usage = pod.get('ephemeral-storage')
+        if pod_ephemeral_usage:
+            used_bytes = pod_ephemeral_usage.get('usedBytes')
+            if used_bytes:
+                self.gauge(self.NAMESPACE + '.ephemeral_storage.usage', used_bytes, pod_tags)
+
+        # Metrics below should already be gathered by another mean (cadvisor endpoints)
+        if not main_stats_source:
+            return
+
+        network = pod.get('network')
+        if network:
+            rx_bytes = network.get('rxBytes')
+            if rx_bytes:
+                self.rate(self.NAMESPACE + '.network.rx_bytes', rx_bytes, pod_tags)
+            tx_bytes = network.get('txBytes')
+            if tx_bytes:
+                self.rate(self.NAMESPACE + '.network.tx_bytes', tx_bytes, pod_tags)
+
+    def _report_container_stats(self, pod_namespace, pod_name, containers, pod_list_utils, instance_tags, main_stats_source):
+        # Metrics below should already be gathered by another mean (cadvisor endpoints)
+        if not main_stats_source:
+            return
+
```

Comment:
should we log something if the containers list is empty?

Reply:
I don't think it'd bring something